### PR TITLE
initial Eltex-mes IPv6 address discovery

### PR DESCRIPTION
**Pull request by Npeca75**
Eltex-mes OS using Radlan MIBs
so half of information is in IP-MIB, other half in RADLAN-IPv6

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Npeca75** [comment]
lnms1
lnms2

**Comment by Jellyfrog** [comment]
mibs should be splitted by vendor instead of os/device

**Comment by Npeca75** [comment]
> mibs should be splitted by vendor instead of os/device

Hi @Jellyfrog 
last time? few month ago? if i remember correctly, PipoCanaja stated that YAML could handle only one "mib dir"

    SNMP['/usr/bin/snmpwalk' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/eltexmes23xx' '-v2c' '-c' 'COMMUNITY' '-OQXUte' 'udp:HOSTNAME:161' 'RADLAN-IPv6::rlIpAddressTable']  

as you see, MIBs are taken from mibs/xxxxx dir

if you have other suggestion could you please post an example ?

File: includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @link       https://www.librenms.org
+ *
+ * @copyright  2021 Peca Nesovanovic
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+ */
+
+//standard MIB
+$oids = SnmpQuery::walk('IP-MIB::ipAddressIfIndex.ipv6')->table(2);
+//Radlan MIB
+$oidr = SnmpQuery::walk('RADLAN-IPv6::rlIpAddressTable')->table(2);
```
Comment (murrant) [inline includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php:29]: You can merge the two arrays by adding the $oids variable as the second argument of ->table().  Will make things easier below.

**Comment by Npeca75** [comment]
> mibs should be splitted by vendor instead of os/device

ok, moved IP and IPv6 MIB in /mibs/radlan and added group: radlan in two Eltex OS
other MIBs could not be simply merged, because they are sooo different, furthermore, mes21 fail discovery with mes23 MIBs and vice versa, so i left them separate. will see if checks fails

File: includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+if ($oids && $oidr) {
+    d_echo('Eltex IPv6: discovering ...');
+
+    foreach ($oids['ipv6'] as $ip => $iparray) {
+        d_echo('Eltex IPv6: processing ' . $ip);
+
+        $index = $iparray['IP-MIB::ipAddressIfIndex'];
+        $prefixlen = $oidr['ipv6'][$ip]['RADLAN-IPv6::rlIpAddressPrefixLength'];
+        $split = str_split(str_replace(':', '', strtolower($ip)), 4); //convert colon delimited hex IPv6 address to array, every forth char
```
Comment (murrant) [inline includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php:39]: We have an IPv6 class to help with ip validation/manipulation.

File: includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php (hunk 1 above)
Comment (Npeca75) [reply includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php]: done

File: includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php (hunk 2 above)
Comment (Npeca75) [reply includes/discovery/ipv6-addresses/eltex-mes21xx.inc.php]: ah, indeed ... :slightly_smiling_face: 
done

**Review by murrant — APPROVED** [review]
Immaculate :)

**Pull request merged**